jina/peapods/runtimes/request_handlers/data_request_handler.py

```diff
@@ -22,7 +23,9 @@ def _get_docs_matrix_from_message(
     field: str,
 ) -> List['DocumentArray']:
     if partial_request is not None:
-        result = [getattr(r, field) for r in reversed(partial_request)]
+        result = [(r.pod_name, getattr(r, field)) for r in reversed(partial_request)]
+        result.sort(key=itemgetter(0))
```

Comment:
why is this needed?

Reply:
This is important to have deterministic results after reducing
It turns out that the document arrays coming from different shards do not have a deterministic order: the one that completes first will be returned first.
As reduce's result (and also match merger) depend on the order of the DocumentArrays, there has to be a way to keep the order deterministic.
I thought about using the pod_name to do so. It's also easy to understand for the user

Reply:
different shards share pod name from the user point of view. This is a weak reason to alter the proto structure so heavily

Reply:
Removed.
Results still depend on the order and are undeterministic. Should be solved in star routing